akka-http/src/main/scala/akka/http/scaladsl/server/PathMatcher.scala

```diff
@@ -358,6 +358,20 @@ trait PathMatchers {
 
   /**
    * A PathMatcher that matches and extracts the complete remaining,
+   * unmatched part of the request's URI path as an decoded (with actual whitespace) String.
+   * @group pathmatcher
+   */
+  object RemainingDecoded extends PathMatcher1[String] {
+    def apply(path: Path) = {
+      Segments(RemainingPath(path).extractions._1) match {
+        case Matched(_, extractions) ⇒ Matched(Path.Empty, Tuple1(extractions._1.mkString))
+        case _ ⇒ Unmatched
+      }
+    }
+  }
```

Comment:
> Wouldn't the same be achievable by doing the decoding here directly, instead of the dance via Segments?
Can you please be more specific? I am not the fan of 

> dance
approach either, but was not able to find anything direct like `decode(path: Path)` :)